### PR TITLE
Add unit test for FlowDirection and rework implementation in ComboBox

### PR DESCRIPTION
**Pull request by danielmayost**
## What does the pull request do?
Add unit test for FlowDirection and rework implementation in `ComboBox`.

`ComboBox` paint his content with a `VisualBrush`, putting the `ComboBoxItem`'s content inside `Rectangle` using a `VisualBrush`.

when the `ComboBox` is closed, the children of `ComboBoxItem` isn't connected to visual tree which causes them to calculate `HasMirrorTransform` as if their parent is LTR, which sometimes wrong, for example if `ComboBox` is RTL, Which will cause the child of ComboBoxItem to apply MirrorTransform while ComboBox has already applied it.

If we set the Rectangle's FlowDirection to LTR, when the content connects to the visual tree (when popup list is open), it recalculates `HasMirrorTransform` according to its parent, and if it is both RTL it does not apply MirrorTransform to itself, but within the `Rectangle` it should do so, because `Rectangle` remains LTR.

The solution I suggest is to give `Rectangle` the `FlowDirection` of parent of `ComboBoxItem`'s content if any, otherwise, LTR.

### CODE REVIEW

File: tests/Avalonia.Controls.UnitTests/ComboBoxTests.cs [hunk 1]
```diff
@@ -336,5 +336,104 @@ public void Close_Window_On_Alt_F4_When_ComboBox_Is_Focus()
...
+                    Items = items,
+                    Template = GetTemplate()
+                };
+
+                var root = new TestRoot(target);
+                target.ApplyTemplate();
+                target.Presenter.ApplyTemplate();
+                target.SelectedIndex = 0;
+
+                // need help here how to connect 'content' tio visual tree, or how to
```
Comment (danielmayost) [inline tests/Avalonia.Controls.UnitTests/ComboBoxTests.cs:430]: @Gillibald

File: tests/Avalonia.Controls.UnitTests/ComboBoxTests.cs [hunk 2]
```diff
@@ -336,5 +336,104 @@ public void Close_Window_On_Alt_F4_When_ComboBox_Is_Focus()
...
+
+                var root = new TestRoot(target);
+                
+                target.ApplyTemplate();
+                target.Presenter.ApplyTemplate();
+                target.SelectedIndex = 0;
+
+                var rectangle = target.GetValue(ComboBox.SelectionBoxItemProperty) as Rectangle;
+
+                // need help here, the 'rectangle' isn't connected to visual tree for some reason
```
Comment (danielmayost) [inline tests/Avalonia.Controls.UnitTests/ComboBoxTests.cs:395]: @Gillibald

File: tests/Avalonia.Controls.UnitTests/ComboBoxTests.cs [hunk 3]
```diff
@@ -336,5 +336,104 @@ public void Close_Window_On_Alt_F4_When_ComboBox_Is_Focus()
...
+                    Items = items,
+                    Template = GetTemplate()
+                };
+
+                var root = new TestRoot(target);
+                target.ApplyTemplate();
+                target.Presenter.ApplyTemplate();
+                target.SelectedIndex = 0;
+
+                // need help here how to connect 'content' to visual tree, or how to open popup
```
Comment (danielmayost) [inline tests/Avalonia.Controls.UnitTests/ComboBoxTests.cs:430]: @Gillibald

Comment (danielmayost) [reply tests/Avalonia.Controls.UnitTests/ComboBoxTests.cs:431]: That's fine, i've done.

**Comment by avaloniaui-team** [comment]
You can test this PR using the following package version. `0.10.999-cibuild0020800-beta`. (feed url: https://nuget.avaloniaui.net/repository/avalonia-all/index.json) [PRBUILDID]

**Comment by danielmayost** [comment]
Ready for review.

**Comment by avaloniaui-team** [comment]
You can test this PR using the following package version. `0.10.999-cibuild0020821-beta`. (feed url: https://nuget.avaloniaui.net/repository/avalonia-all/index.json) [PRBUILDID]

**Comment by avaloniaui-team** [comment]
You can test this PR using the following package version. `0.10.999-cibuild0020861-beta`. (feed url: https://nuget.avaloniaui.net/repository/avalonia-all/index.json) [PRBUILDID]

**Review by Gillibald — APPROVED** [review]

**Comment by avaloniaui-team** [comment]
You can test this PR using the following package version. `0.10.999-cibuild0021258-beta`. (feed url: https://nuget.avaloniaui.net/repository/avalonia-all/index.json) [PRBUILDID]

**Pull request merged**